src/Razor/test/Microsoft.AspNetCore.Razor.Test.Common/Language/IntegrationTests/IntegrationTestBase.cs

```diff
@@ -519,6 +519,70 @@ protected void AssertSourceMappingsMatchBaseline(RazorCodeDocument codeDocument)
             }
         }
 
+        protected void AssertLinePragmas(RazorCodeDocument codeDocument, bool designTime)
+        {
+            if (FileName == null)
+            {
+                var message = $"{nameof(AssertSourceMappingsMatchBaseline)} should only be called from an integration test. ({nameof(FileName)} is null).";
+                throw new InvalidOperationException(message);
+            }
+
+            var csharpDocument = codeDocument.GetCSharpDocument();
+            Assert.NotNull(csharpDocument);
+            var linePragmas = csharpDocument.LinePragmas;
+            designTime = false;
+            if (designTime)
+            {
+                var sourceMappings = csharpDocument.SourceMappings;
+                foreach (var sourceMapping in sourceMappings)
+                {
+                    var foundMatchingPragma = false;
+                    foreach (var linePragma in linePragmas)
+                    {
+                        if (sourceMapping.OriginalSpan.LineIndex >= linePragma.StartLineIndex &&
+                            sourceMapping.OriginalSpan.LineIndex <= linePragma.EndLineIndex)
+                        {
+                            // Found a match.
+                            foundMatchingPragma = true;
+                            break;
+                        }
+                    }
+
+                    Assert.True(foundMatchingPragma, $"No line pragma found for code at line {sourceMapping.OriginalSpan.LineIndex + 1}.");
+                }
+            }
+            else
+            {
+                var syntaxTree = codeDocument.GetSyntaxTree();
+                var sourceBuffer = new char[syntaxTree.Source.Length];
+                syntaxTree.Source.CopyTo(0, sourceBuffer, 0, syntaxTree.Source.Length);
+                var sourceContent = new string(sourceBuffer);
+                var classifiedSpans = syntaxTree.GetClassifiedSpans();
+                foreach (var classifiedSpan in classifiedSpans)
+                {
+                    var content = sourceContent.Substring(classifiedSpan.Span.AbsoluteIndex, classifiedSpan.Span.Length);
+                    if (!string.IsNullOrWhiteSpace(content) &&
+                        classifiedSpan.BlockKind != BlockKindInternal.Directive &&
```

Comment:
Excluding directives here because there are some inconsistencies here and there. E.g,
For `@section SectionName { ... }`, `SectionName` is categorized as code and there is no corresponding line pragma in generated code. Similarly for `@implements`. I'll work on fixing them incrementally after this PR.

Reply:
It seems good for section. It seems like we want to fix things like `implements`.